### ISSUE TITLE
[valid-expect-in-promise] Problem chaining .then's

### ISSUE THREAD

**Issue by pladaria** [jest-community/eslint-plugin-jest#405]
using eslint-plugin-jest v22.15.2

This reports an error:
```javascript
const fetch = url =>
    Promise.resolve({
        url,
        text: () => Promise.resolve('text'),
    });

// Error: Promise should be returned to test its fulfillment or rejection
test('valid-expect-in-promise', async () => {
    const text = await fetch('url')
        .then(res => res.text()) 
        .then(text => text);     

    expect(text).toBe('text');
});
```
If I remove the last `.then`, no error is reported:
```javascript
// No eslint errors
test('valid-expect-in-promise', async () => {
    const text = await fetch('url')
        .then(res => res.text()) 

    expect(text).toBe('text');
});
```
The following doesn't report error, so perhaps the problem is not in chaining:
```javascript
// No eslint errors
test('valid-expect-in-promise', async () => {
    const text = await fetch('url')
        .then(text => text)
        .then(text => text);

    expect(text).toBe('text');
});
```

**Comment by SimenB** [comment]
@tushardhole any idea?

**Comment by pladaria** [comment]
Cannot reproduce in 22.15.0

**Comment by G-Rath** [comment]
```
Index: src/rules/__tests__/valid-expect-in-promise.test.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/rules/__tests__/valid-expect-in-promise.test.ts	(revision 0cb1df57d1d524150abf6452d50dceec82f60707)
+++ src/rules/__tests__/valid-expect-in-promise.test.ts	(date 1567074887550)
@@ -10,6 +10,19 @@
 ruleTester.run('valid-expect-in-promise', rule, {
   valid: [
     `
+    const fetch = url => Promise.resolve({
+      url, text: () => Promise.resolve('text')
+    });
+
+    test('valid-expect-in-promise', async () => {
+      const text = await fetch('text')
+        .then(res => res.text())
+        .then(text => text);
+
+      expect(text).toBe('text');
+    })
+    `,
+    `
       it('it1', () => new Promise((done) => {
         test()
           .then(() => {
Index: src/rules/valid-expect-in-promise.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/rules/valid-expect-in-promise.ts	(revision 0cb1df57d1d524150abf6452d50dceec82f60707)
+++ src/rules/valid-expect-in-promise.ts	(date 1567074872055)
@@ -180,7 +180,11 @@
           return;
         }
         const testFunction = findTestFunction(node);
-        if (testFunction && !isHavingAsyncCallBackParam(testFunction)) {
+        if (
+          testFunction &&
+          !isHavingAsyncCallBackParam(testFunction) &&
+          !testFunction.async
+        ) {
           const { body } = testFunction;
 
           /* istanbul ignore if https://github.com/typescript-eslint/typescript-eslint/issues/734 */
```

**Comment by DanielSWolf** [comment]
Has there been any progress? I'm having the same problem with false positives.
